src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    public T[] grow(T value, int index) {
+        T[] newElementData = null;
+        if (size < elementData.length) {
+            newElementData = (T[]) new Object[elementData.length];
+        }
+        if (size == elementData.length) {
+            newElementData = (T[]) new Object[elementData.length + elementData.length / 2];
+        }
+        System.arraycopy(elementData, 0, newElementData, 0, index);
+        newElementData[index] = value;
+        System.arraycopy(elementData, index, newElementData, index + 1, size - index);
+        elementData = newElementData;
+        size++;
+        return elementData;
+    }
+
+    public void indexCheck(int index) {
```

Comment:
```suggestion
    private  void indexCheck(int index) {
```
this method should be private, and move it after public